compiler/optimizer/abstractinterpreter/OMRIDTBuilder.cpp

```diff
@@ -0,0 +1,324 @@
+   TR::IDTNode* interpretedMethodIDTNode = checkIfMethodIsInterpreted(method);
+
+   if (interpretedMethodIDTNode)
+      {
+      // since they are the same method, they should have the same inlining method summary.
+      node->setInliningMethodSummary(interpretedMethodIDTNode->getInliningMethodSummary());
+      uint32_t staticBenefit = computeStaticBenefit(node->getInliningMethodSummary(), arguments);
+
+      node->setStaticBenefit(staticBenefit);
+
+      // Same methods should have same IDT decendents.
+      // IDT is built in DFS order, at this point, we have all the descendants so it is safe to copy all the descendants.
+      copyDescendants(interpretedMethodIDTNode, node);
+      return;
+      }
+   else
+      {
+      // Abstract interpretation will identify and find callsites thus they will be added to the IDT
+      TR::IDTBuilderVisitor visitor(self(), node, nextCallStack);
+
+      self()->performAbstractInterpretation(node, visitor, arguments, callerIndex);
+
+      // At this point we have the inlining summary generated by abstract interpretation
+      // So we can use the summary and the arguments passed from callers to calculate the static benefit.
+      if (!node->isRoot())
+         {
+         uint32_t staticBenefit = computeStaticBenefit(node->getInliningMethodSummary(), arguments);
+         node->setStaticBenefit(staticBenefit);
+         }
+
+      // Save for later use if we encounter any same method.
+      storeInterpretedMethod(method, node);
+      }
+   }
+
+void OMR::IDTBuilder::copyDescendants(TR::IDTNode* fromNode, TR::IDTNode* toNode)
+   {
+   TR_ASSERT_FATAL(
+      fromNode->getResolvedMethodSymbol()->getResolvedMethod()->getPersistentIdentifier()
+      == toNode->getResolvedMethodSymbol()->getResolvedMethod()->getPersistentIdentifier(),
+      "Copying different nodes is not allowed!");
+
+   for (int32_t i = 0 ; unsigned(i) < fromNode->getNumChildren(); i ++)
+      {
+      TR::IDTNode* child = fromNode->getChild(i);
+
+      if (toNode->getBudget() - child->getCost() < 0)
+         continue;
+
+      if (toNode->getRootCallRatio() * child->getCallRatio() < COLD_ROOT_CALL_RATIO)
+         continue;
+
+      TR::IDTNode* copiedChild = toNode->addChild(
+                           _idt->getNextGlobalIDTNodeIndex(),
+                           child->getCallTarget(),
+                           child->getResolvedMethodSymbol(),
+                           child->getByteCodeIndex(),
+                           child->getCallRatio(),
+                           _idt->getRegion()
+                           );
+
+      if (copiedChild)
+         {
+         _idt->addCost(copiedChild->getCost());
+         _idt->increaseGlobalIDTNodeIndex();
+         copiedChild->setInliningMethodSummary(child->getInliningMethodSummary());
+         copiedChild->setStaticBenefit(child->getStaticBenefit());
```

Comment:
Same as https://github.com/eclipse/omr/pull/5509#discussion_r1376493605, I will take a look for this and give you feedback soon later.

Reply:
I think https://github.com/eclipse/omr/pull/5509#discussion_r1376493605 may also answer this question. The arguments won't affect the static benefit.

Reply:
I think that normally the static benefit would be determined by `computeStaticBenefit()` in a way that does depend on the arguments

Reply:
Yes, the static benefit should not simply copied. However, I didn't find a way to get arguments without running abstract interepretation, because the arguments are updated by abstract interepretation. More details are at https://github.com/eclipse/omr/pull/5509#discussion_r1384233283.